spec/solver_spec.rb

```diff
@@ -0,0 +1,45 @@
+require_relative '../solver'
+
+describe Solver do
+  solver = Solver.new
+  context 'when calculating factorial' do
+    it 'Should return 1' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'should return the real answer' do
+      num = 5
+      expect(solver.factorial(num)).to eq(120)
+    end
+  end
```

Comment:
- Dear Team, You did great with the project, but here we need to add a test for a negative number. Remember we need to have a test for a negative value which raises an exception as an expectation. Please add this test. 🙏